var/spack/repos/builtin/packages/py-cellprofiler/package.py

```diff
@@ -0,0 +1,153 @@
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyCellprofiler(PythonPackage):
+    """CellProfiler cell image analysis software.
+
+    CellProfiler is a free open-source software designed to enable biologists
+    without training in computer vision or programming to quantitatively
+    measure phenotypes from thousands of images automatically.
+
+    """
+
+    homepage = "https://cellprofiler.org"
+    pypi = "cellprofiler/CellProfiler-4.2.6.tar.gz"
+    git = "https://github.com/CellProfiler/CellProfiler.git"
+
+    maintainers("omsai")
+
+    license("BSD-3-Clause", checked_by="omsai")
+
+    version("4.2.6", sha256="37e2a35dccff456afda96a4442dff2d23809c8ee271607a347e386aeb4af2628")
+
+    depends_on("python@3.8:", type=("build", "run"))
+
+    depends_on("py-setuptools@64:", type="build")
+    depends_on("py-setuptools-scm@8:", type="build")
+
+    depends_on("py-boto3@1.12.28:", type=("build", "run"))
+    depends_on("py-cellprofiler-core@4.2.6", type=("build", "run"))
+    depends_on("py-centrosome@1.2.2:", type=("build", "run"))
+    depends_on("py-docutils@0.15.2:", type=("build", "run"))
+    # More recent versions of h5py cause:
+    #   AttributeError: module 'h5py' has no attribute 'Dataset
+    depends_on("py-h5py@3.6:3.7~mpi", type=("build", "run"))
+    depends_on("py-imageio@2.5:", type=("build", "run"))
+    depends_on("py-inflect@2.1:6", type=("build", "run"))
+    depends_on("py-jinja2@2.11.2:", type=("build", "run"))
+    depends_on("py-joblib@0.13:", type=("build", "run"))
+    depends_on("py-mahotas@1.4:", type=("build", "run"))
+    # matplotlib.cm.get_cmap does not exist in 3.9.0 onwards.
+    depends_on("py-matplotlib@3.1.3:3.8", type=("build", "run"))
+    depends_on("py-mysqlclient@1.4.6", type=("build", "run"))
+    depends_on("py-numpy@1.20.1:", type=("build", "run"))
+    depends_on("py-pillow@7.1:", type=("build", "run"))
+    depends_on("py-prokaryote@2.4.4:", type=("build", "run"))
+    depends_on("py-python-bioformats@4.0.7:", type=("build", "run"))
+    depends_on("py-python-javabridge@4.0.3:", type=("build", "run"))
+    depends_on("py-pyzmq@22.3:22", type=("build", "run"))
+    depends_on("py-sentry-sdk@0.18:", type=("build", "run"))
+    depends_on("py-requests@2.22:", type=("build", "run"))
+    depends_on("py-scikit-image@0.18.3:", type=("build", "run"))
+    depends_on("py-scikit-learn@0.20:0", type=("build", "run"))
+    depends_on("py-scipy", type=("build", "run"))
+    depends_on("py-six", type=("build", "run"))
+    depends_on("py-tifffile@:2022.4.21", type=("build", "run"))
+    depends_on("py-wxpython@4.1.0:4", type=("build", "run"))
+
+    depends_on("py-pytest", type=("run", "test"), when="@4.2.6 +tests")
+
+    # Run the post-install tests with `spack test run py-cellprofiler`.  We
+    # need the variant to add the pytest executable to the PATH.
+    variant("tests", default=False, description="Post-install tests.")
```

Comment:
I don't quite understand the need for a variant. Doesn't the spack testing functionality already address this. By specifying that `py-pytest` is a dependency of type `test`, it will be required when the users does something like `spack install --test root py-cellprofiler`. That's the preferred approach over variants to control this behavior. See more at https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#testing.

Reply:
Per https://github.com/spack/spack/pull/44824/files#r1740396700 going with the `tests` variant is just fine here. Resolving.